src/routes/github/create-branch/github-create-branch-post.ts

```diff
@@ -5,20 +5,56 @@ import { AnalyticsEventTypes, AnalyticsTrackEventsEnum } from "interfaces/common
 import { statsd } from "~/src/config/statsd";
 import { metricCreateBranch } from "~/src/config/metric-names";
 import { getCloudOrServerFromGitHubAppId } from "~/src/util/get-cloud-or-server";
+import { Subscription } from "models/subscription";
 
-const errorMessages = {
-	// TODO: Fix the url later, once you figure out how to get the `installationId`
-	403: "We couldn’t create this branch, possibly because this GitHub repository hasn't been configured to your Jira site. <a href='#'>Allow access to this repository.</a>",
-	400: "We couldn’t create this branch. Check that you’ve entered valid values for repository, source branch name, and new branch name.",
-	422: "We couldn’t create this branch, possibly because:<ul>" +
-		"<li>This GitHub branch already exists. Please try a different branch name.</li>" +
-		"<li>The GitHub branch name is not valid.</li>" +
-		"</ul>",
-	404: "We couldn’t create this branch, possibly because the GitHub source branch you entered doesn't exist. Please try a different branch.",
-	500: "We couldn’t create this branch because something went wrong. Please try again."
+const errorMessages = (statusCode: number, url?: string): string => {
+	switch (statusCode) {
+		case 403: {
+			return `We couldn’t create this branch, possibly because this GitHub repository hasn't been configured to your Jira site. <a href="${url}" target="_blank">Allow access to this repository.</a>`;
```

Comment:
I'm taking ages to review because I keep coming back to the switch statement....
Id like another opinion I'm on the fence.

I prefer the original format here,  I understand why we changed it but perhaps we leave this as it was and then interpolate the result. Changing the structure of this to solve logic somewhere else too heavy handed.

e.g. if it was all still 403: "blah blah blah maybe blah {url}"...

Then in the if statement below
```
	if (err.status === 403) {
			const gitHubConfigurationLink = await getGitHubConfigurationLink(gitHubAppConfig, gitHubUserClient, jiraHost, owner, repo);
			const forbiddenErrorYo = genericUtilReplaceStringfunctionOfNice(errorMessages(err?.status || 500, gitHubConfigurationLink) // this replaces ? with ...args
						res.status(err.status).json(forbbinErrorYo);
```

Reply:
The original format with the object mapping was easier. But that means we have to modify the message again if its a 403. I prefer getting the modified or unmodified message from the same method, so I think better to use this method.